sys/net/application_layer/nanocoap/nanocoap.c

```diff
@@ -595,32 +605,106 @@ size_t coap_put_option_uri(uint8_t *buf, uint16_t lastonum, const char *uri, uin
     return bufpos - buf;
 }
 
+size_t coap_block2_init(uint8_t* buf, uint16_t lastonum, coap_pkt_t *pkt, coap_blockhelper_t *blk)
+{
+    uint32_t blknum;
+    unsigned szx;
+    if(coap_get_blockopt(pkt, COAP_OPT_BLOCK2, &blknum, &szx) >= 0) {
+        /* Use the smallest block size */
+        if (NANOCOAP_BLOCK_SZX_MAX - 4 < szx ) {
```

Comment:
I'm confused, what does 'smallest' mean here? The smallest value of szx is 0. Also, watch the spacing in l.614 and 615.

Reply:
I'll clarify the comment here. This should describe that the block size used is the smallest block size among what the client requested and the maximum that nanocoap can handle. ([rfc7959 - sec 2.4 second paragraph](https://tools.ietf.org/html/rfc7959#section-2.4))

Reply:
See comment above on definition of NANOCOAP_BLOCK_SZX_MAX in nanocoap.h. We can resolve there.